es6/experiment.js

```diff
@@ -1,213 +1,214 @@
-import Assignment from './assignment';
 import { shallowCopy, extend, isObject, forEach, map, trimTrailingWhitespace } from './lib/utils';
 
-class Experiment {
-  constructor(inputs) {
-    this.inputs = inputs;
-    this._exposureLogged = false;
-    this._salt = null;
-    this._inExperiment = true;
+export default function provideExperiment(Assignment) {
+  class Experiment {
+    constructor(inputs) {
+      this.inputs = inputs;
+      this._exposureLogged = false;
+      this._salt = null;
+      this._inExperiment = true;
 
-    this.name = this.getDefaultExperimentName();
-    this._autoExposureLog = true;
+      this.name = this.getDefaultExperimentName();
+      this._autoExposureLog = true;
 
-    this.setup();
+      this.setup();
 
-    this._assignment = new Assignment(this.getSalt());
-    this._assigned = false; 
-  }
+      this._assignment = new Assignment(this.getSalt());
+      this._assigned = false;
+    }
 
-  //helper function to return the class name of the current experiment class
-  getDefaultExperimentName() {
-    if (isObject(this) && this.constructor && this !== this.window) {
-      var arr = this.constructor.toString().match(/function\s*(\w+)/);
-      if (arr && arr.length === 2) {
-        return arr[1];
+    //helper function to return the class name of the current experiment class
+    getDefaultExperimentName() {
```

Comment:
While going through this PR, this line caught my attention and reminded me of an issue that we've only ever documented internally at hubspot.

I put up a public issue here - https://github.com/HubSpot/PlanOut.js/issues/57

This issue might require a major version bump. If timing allows, I'd like to package the changes in this PR with the changes from https://github.com/HubSpot/PlanOut.js/issues/57 into a single v4.0 release.